### ISSUE TITLE
Slice Geometry throws NullPointerExceptions when calculating Thickness values

### ISSUE THREAD

**Issue by alessandrofelder** [bonej-org/BoneJ2#193]
**Bug description**
When running Slice Geometry on a stack of binary images, NullPointerExceptions are thrown in some threads.

```
Exception in thread "Thread-29307" java.lang.NullPointerException
	at sc.fiji.localThickness.EDT_S1D.run(EDT_S1D.java:208)
	at sc.fiji.localThickness.LocalThicknessWrapper.processImage(LocalThicknessWrapper.java:127)
	at org.bonej.menuWrappers.LocalThickness.processThicknessSteps(LocalThickness.java:266)
	at org.bonej.menuWrappers.LocalThickness.getLocalThickness(LocalThickness.java:163)
	at org.bonej.plugins.SliceGeometry$SliceThread.run(SliceGeometry.java:1005)
```

**To Reproduce**
With the latest Fiji and BoneJ2 installed, 
1. Open the test image provided [in this forum post](https://forum.image.sc/t/problem-with-bonej2-slice-geometry/29214)
2. Run Plugins > BoneJ > Slice Geometry
3. Tick "Process Stack", and then OK
4. Log shows NullPointerExceptions
5. Results table column Mean3DThickness has lots of zero and NaN entry

**Expected behavior**
No NullPointerExceptions, mean thickness values are >0, finite, and the same for each slice (in this test image)

**Environment**
 - Ubuntu 16.04
 - java 1.8.0_171

**Comment by mdoube** [comment]
This looks like a problem in the reimplementation of Local Thickness and the way it handles 2D slices, or perhaps the way that Local Thickness (complete process) is called slice by slice on a stack by Slice Geometry in BoneJ2. It might be some kind of thread collision, because Thickness itself is multithreaded and so might be the slice processing in Slice Geometry.

This bug does not exist in BoneJ1.

**Comment by mdoube** [comment]
> Results table column Mean3DThickness has lots of zero and NaN entry

This is probably due to a previous run of Slice Geometry creating a 3D thickness column, which is then filled with dummy entries because the 3D option is not selected in a subsequent run. Probably not a symptom of this bug.

**Comment by rimadoma** [comment]
@mdoube Maybe you added something to Thickness in BoneJ1 that allowed it processing 2D (single slice) images? Something that isn't in the original `sc.fiji.LocalThickness_` plugin the implementation now calls. That's where the `NullPointerException` happens at least.

**Comment by rimadoma** [comment]
This error goes as far back as `LocalThickness` has been in this repo (commit 4dfff5). Unfortunately the first commit of this repo wasn't a clone of BoneJ1 as is. In retrospect that was a big mistake. We'll have to compare `Thickness` from mdoube's BoneJ1 repo to `sc.fiji.LocalThickness_` to see what's different. 

In `Legacy` there's also a deprecated `LocalThickness`, which is there to allow other plugins such as `SliceGeometry` to call features as before without having to worry about `sc.fiji.LocalThickness_`. The error might also come from there.

**Comment by rimadoma** [comment]
Solved in PR https://github.com/fiji/LocalThickness/pull/7. Local thickness wasn't thread safe, because the sub-plugins (Distance ridge etc) were `static`.

If you want to test this
1. Pull the PR from LocalThickness, 
2. Run `mvn install` to add `LocalThickness_:4.0.3-SNAPSHOT` to your local Maven repository
3. Make `bonej-legacy-plugins_` `POM` depend on `LocalThickness_:4.0.3-SNAPSHOT`
4. Install with ` mvn install -Dscijava.app.directory=/c/Fiji.app/ -Denforcer.skip=true` (enforcer complains about SNAPSHOT dependencies)
`

**Comment by ctrueden** [comment]
PR is merged. New snapshot should build and deploy shortly, which will eliminate the need for steps (1) and (2).

**Comment by ctrueden** [comment]
`sc.fiji:LocalThickness_:4.0.3` release is now available. Will be part of pom-scijava 29.0.0 (release target before end of March). In the meantime, you could update BoneJ2 to include `<LocalThickness_.version>4.0.3</LocalThickness_.version>` in `<properties>`.

**Comment by mdoube** [comment]
@ctrueden is 5143f40 the right way to do it, or is there more to do somewhere else?

**Comment by ctrueden** [comment]
@mdoube Yep, that's exactly the right way. :smile:

**Comment by mdoube** [comment]
Solved again by #213
